core/lib/vm_interface/src/types/errors/halt.rs

```diff
@@ -42,6 +42,7 @@ pub enum Halt {
     VMPanic,
     TracerCustom(String),
     FailedToPublishCompressedBytecodes,
+    FailedBlockTimestampAssertion,
```

Comment:
Future work (?): AFAICT, right now this error is only raised outside the VM, by `MempoolIO`. Is it possible to convert actual `require` failure in the timestamp asserter to this variant? Or would it require too much work?

Reply:
Thanks. Created a ticket for future investigation.